main.c

```diff
@@ -24,6 +26,32 @@ static struct option options[] = {
 
 /* ------------------------------------------------------------------------ */
 
+static void
+set_timespec(struct timespec *tv, unsigned long long int msecs)
+{
+	tv->tv_sec = msecs / 1000;
+	tv->tv_nsec = (msecs % 1000) * 1000000;
+}
+
+/* ------------------------------------------------------------------------ */
+
+static int
+wait_signalfd(int sigfd, struct timespec *tv)
+{
+	int ret;
+	fd_set fdset;
+	FD_ZERO(&fdset);
+	FD_SET(sigfd, &fdset);
```

Comment:
Right now there is no chance but I suppose after some future changes that might not be true. It shouldn't hurt much to put if() around it so I'll do it. [It would be cool if compiler was smart enough to optimize that away when it's not needed :smile: but I doubt that.]